src/main/java/org/commcare/formplayer/application/MenuController.java

```diff
@@ -168,8 +168,10 @@ public BaseResponseBean navigateSessionWithAuth(@RequestBody SessionNavigationBe
             @CookieValue(Constants.POSTGRES_DJANGO_SESSION_ID) String authToken,
             HttpServletRequest request) throws Exception {
         String[] selections = sessionNavigationBean.getSelections();
-        MenuSession menuSession;
-        menuSession = menuSessionFactory.getMenuSessionFromBean(sessionNavigationBean);
+        if (selections == null) {
+            selections = new String[0];
```

Comment:
This makes sense to me, but could this cause any problems in the case of `nonAppNav` in `advanceSessionWithSelections`? I'm not super familiar with when/how `nonAppNav` would be true, but if `nonAppNav` was true the length of `selections` was `0`,  `selections = Arrays.copyOf(selections, selections.length - 1)` might throw an error.

Reply:
That's a good point, I think `null` selections are only sent as part of `navigate_menu_start` but nevertheless added [a empty selections check](https://github.com/dimagi/formplayer/pull/1590/commits/5136e3e0f1000a26e4b47256b31aaf0da3f7b2da) for safety here.